al_util.js

```diff
@@ -2,18 +2,43 @@
  * @copyright (C) 2017, Alert Logic, Inc
  * @doc
  *
- * Helper utilities for  Alert Logic log collector.
+ * Helper utilities for  Alert Logic collectors.
  *
  * @end
  * -----------------------------------------------------------------------------
  */
-'use strict';
 
 const rp = require('request-promise-native');
-const path = require('path');
+const retry = require('retry');
 
 let MAX_CONNS_PER_SERVICE = 128;
 
+/**
+ * @default Refer to https://www.npmjs.com/package/retry
+ */
+let DEFAULT_RETRY = {
+    // Default values
+    // factor: 2,
+    // randomize: false
+    minTimeout: 3000,
+    retries: 2,
+    maxTimeout: 30000
```

Comment:
Isn't it too much of a delay for lambda?

Reply:
Have just 2 retries here means that there will be one retry after 3 secs and another one after 9 secs.
By default, `maxTimeout` is set is `Infinity`. Just want to make sure that is anybody increases `retries` the maximum timeout would not exceed 30 secs.
Obviously we can change default retry strategy to something like:
```
    minTimeout: 1000,
    retries: 3,
    maxTimeout: 10000
```
Anyway the caller function is able to set its own retry strategy and here is just my best guess for default values. Any ideas are welcomed.

Reply:
I think we can reduce such timeouts to smaller values as advised by @motobob to something like 300/600/900

Reply:
Default retry timeouts changes to 300/2100/10000